railties/lib/rails/generators/rails/app/templates/config/boot.rb.tt

```diff
@@ -1,13 +1,13 @@
-<% unless options.skip_spring? -%>
-begin
+# frozen_string_literal: true
+
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../Gemfile', __dir__)
+<% if spring_install? -%>
+
+if !defined?(Spring) && [nil, "development", "test"].include?(ENV["RAILS_ENV"])
   load File.expand_path("../bin/spring", __dir__)
```

Comment:
If we're generating this file ourselves, it seems odd to continue to put it in `bin/`, when it's only ever going to be loaded from a file in `config/`. Or is the idea to let applications use it as a "regular" Spring binstub later?

Reply:
`bin/spring` is still a valid command. It is there so people can write custom spring commands and call them. `bin/spring rails` for example works. That file is a binstube. I don't think it should be in the config folder.

Reply:
> `bin/spring rails` for example works

TIL! It seemed to me like the goal of https://github.com/rails/rails/pull/39225 was to make the Spring binstub unnecessary, but you're right that custom commands will still need it. To be honest I'm not totally clear on what we're trying to achieve with these changes, so I might bow out of reviewing this one 😬